node/src/index.ts

```diff
@@ -176,16 +176,21 @@ export async function connect (
     opts = { uri: arg }
   } else {
     // opts = { uri: arg.uri, awsCredentials = arg.awsCredentials }
-    opts = Object.assign(
-      {
-        uri: '',
-        awsCredentials: undefined,
-        awsRegion: defaultAwsRegion,
-        apiKey: undefined,
-        region: defaultAwsRegion
-      },
-      arg
-    )
+    const keys = Object.keys(arg)
+    if (keys.length === 1 && keys[0] === 'uri' && typeof arg.uri === 'string') {
+      opts = { uri: arg.uri }
+    } else {
+      opts = Object.assign(
+        {
+          uri: '',
+          awsCredentials: undefined,
+          awsRegion: defaultAwsRegion,
+          apiKey: undefined,
+          region: defaultAwsRegion
+        },
+        arg
```

Comment:
you're obviously more familiar with this code than I am, but I would have thought that the best solution here would be to explicitly check the URL type & follow two different code paths, like `RemoteConnection` below.

Reply:
(or [perhaps edit the rust part to not try and validate aws creds](https://github.com/lancedb/lancedb/blob/main/rust/ffi/node/src/lib.rs#L205-L240) when it doesn't receive an S3 url)

Reply:
TBH we are a few weeks away from stopping development of this SDK in favor of the new Node one, so I'm not inclined to put too much effort here. This seemed like the easiest quick fix.